wd_cipher.c

```diff
@@ -323,19 +321,14 @@ static int wd_cipher_common_init(struct wd_ctx_config *config,
 
 static void wd_cipher_common_uninit(void)
 {
-	void *priv = wd_cipher_setting.priv;
-
-	if (!priv)
-		return;
-
```

Comment:
There is a problem when deleting directly here. After the priv is removed at the API layer, a solution needs to be added to prevent repeated initialization. Otherwise, the function of preventing double release will be gone, which will cause other problems.

Thanks.
Longfang.

Reply:
The priv here is still in the wd_cipher_setting structure. It is used to prevent repeated uninitialization of resources at the API layer

Reply:
If the driver does not support repeated exit, it is the driver's responsibility to prevent repeated un-initialization, not the caller's.
Now both hisi_sec_init and  hisi_sec_exit has checked drv.priv.
For drivers permitted multi-init and multi-exit, it is OK, just like free().

Reply:
But this problem is not solved in the current patch. If merged into the patch will cause problems

Reply:
wd_alg_try_init(&wd_digest_setting.status) can prevent re-init2
Whether the driver can call init multi-times depends on the driver itself.

Reply:
Yes, this wd_alg_try_init is possible. 
You can add it and verify it.

Reply:
wd_alg_try_init still there, in wd_xxx_init, this patch does not change that.